### DISCUSSION TITLE
memory settings for big database or just too much returned data?

### DISCUSSION THREAD

**Discussion by misu200** [orientechnologies/orientdb#9429] [Q&A]
Hello,

I will describe some behavior I have with a 8 GB database ( 10 million nodes & 10 million edges  ) that I cant explain myself ( OrientDB 3.0.34/Ubuntu/Java 8) . Xms=4gb,Xmx=12gb  on a machine with 32 cores & 64gb ram

I am periodically executing something like 3 million traverse SQL query in OrientDB with a depth=4 by OrientDB HTTP API and getting the subgraph  back

The biggest problem I have is random speed variations from 1 query per 5 minute to like 100 query per second. It goes like this:
5 minute I get 3 query/s .... after I can get 30query/s for 10m ......the 100/s for 20m ..then back to 0 query/s for a few minutes ..and so on

Thee query looks like this:
traverse * from (select * from Firma where cui="13990040")  while ($depth <=4 and (@class="Firma" or @class="Persoana" or @class="Actionar" or @class="Administrator")) strategy BREADTH_FIRST

Its true the query can return sometimes a few hundred kb ..but usually returns just a few kb

Do you have any idea what should I try?

Thanks,
Mihai

**Comment by andrii0lomakin** [comment]
Hm @luigidellaquila , AFAIK match queries always better than traverse queries or did I miss something?

**Reply by luigidellaquila** [reply]
@laa In this case I don't expect much difference.

@misu200 do you see a pattern in the execution time? eg. is the same query getting different performance depending on when you run it, or are different (but similar) queries having different performance?
Maybe the slowdown happens under different load (eg. during write operations)?

Thanks

Luigi

**Reply by misu200** [reply]
I don't do writes in OrientDB . I am happy to use it just for more or less complex traversals and I find it usually very powerful for my needs .) I do use and like  MATCH a lot and sometimes Gremlin when I can't do it with MATCH

I do get like 30 queries/s done from 4 threads in parallel for my 3 millions queries. ( increasing the number of threads doesn't help me from my tests)
However stuff like 1(or 0 ) query/s for 10 minutes and then 100 query/s for the next 10 minutes it puzzles me.

**Reply by luigidellaquila** [reply]
Do you have a chance to take a thread dump while the queries are stuck? It will likely give us significant info to understand the problem

Thanks

Luigi

**Reply by misu200** [reply]
Yes  @luigidellaquila ...I ve attached the tread dump here

Thanks,
Mihai
[thread_dump.txt](https://github.com/orientechnologies/orientdb/files/5466919/thread_dump.txt)
